### PR TITLE
display kind 6 repost impl

### PR DESCRIPTION
**Pull request by kernelkind**
display kind 6 reposts

### CODE REVIEW

File: src/ui/note/mod.rs [hunk 1]
```diff
@@ -415,6 +455,36 @@ impl<'a> NoteView<'a> {
     }
 }
 
+fn get_reposted_note_key(ndb: &Ndb, txn: &Transaction, note: &Note) -> Option<NoteKey> {
+    let new_note_id: &[u8; 32] = if note.kind() == 6 {
+        let mut res = None;
+        for tag in note.tags().iter() {
+            if let Some(tag_str) = tag.get(0).unwrap().variant().str() {
```
Comment (jb55) [inline src/ui/note/mod.rs:462]:
```suggestion
            if let Some("e") = tag.get_unchecked(0).variant().str() {
```

File: src/ui/note/mod.rs [hunk 2]
```diff
@@ -415,6 +455,36 @@ impl<'a> NoteView<'a> {
     }
 }
 
+fn get_reposted_note_key(ndb: &Ndb, txn: &Transaction, note: &Note) -> Option<NoteKey> {
```
Comment (jb55) [inline src/ui/note/mod.rs:458]: you can just return the note here

```suggestion
fn get_reposted_note<'a>(ndb: &Ndb, txn: &'a Transaction, note: &Note) -> Option<Note<'a>> {
```

File: src/ui/note/mod.rs [hunk 3]
```diff
@@ -415,6 +455,36 @@ impl<'a> NoteView<'a> {
...
+    let new_note_id: &[u8; 32] = if note.kind() == 6 {
+        let mut res = None;
+        for tag in note.tags().iter() {
```
Comment (jb55) [inline src/ui/note/mod.rs:461]:
```suggestion
        for tag in note.tags().iter() {
            if tag.count() == 0 {
	        continue;
            }
```

this is needed or we can crash on `unwrap` or `get_unchecked` with a malformed note, but maybe we should just do this so its safe even during refactors:

```rust
if let Some("e") = tag.get(0).and_then(|t|t.variant().str())
```

**Review by jb55 — CHANGES_REQUESTED** [review]
A few comments!

**Comment by kernelkind** [comment]
rebased with master

**Pull request merged**